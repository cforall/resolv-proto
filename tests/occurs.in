// based on "occurs check" example on p. 26 of Richard's thesis

f #fn< T #ptr<T> >
#fn< #ptr<U> U > g

%%

// should fail to resolve
f( g() )