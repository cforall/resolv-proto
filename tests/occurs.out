ERROR: no valid resolution for f( g( ) )
